### ISSUE TITLE
Error when opening some Battles

### ISSUE THREAD

**Issue by ralpha** [Kromtec/LegendsViewer#50]
Steps to reproduce bugs:
- Look in Warfare -> Battles -> Search
- Select a battle from the list
- Error pops up (in ~50% of items clicked on)
(clicking 'Continue' does not quit program, but does not load the battle info)

Info:
- Version used: 1.20.02
- DF version: 0.47.02

Here is the debug info:
```
See the end of this message for details on invoking 
just-in-time (JIT) debugging instead of this dialog box.

************** Exception Text **************
System.NullReferenceException: Object reference not set to an instance of an object.
   at LegendsViewer.Controls.Map.MapPanel..ctor(Bitmap map, World world, DwarfTabControl dwarfTabControl, Object focusObject) in C:\Users\Thomas Svoboda\source\repos\Kromtec\LegendsViewer\LegendsViewer\Controls\Map\MapPanel.cs:line 288
   at LegendsViewer.Controls.Map.MapPanel.CreateBitmaps(World world, DwarfObject dwarfObject) in C:\Users\Thomas Svoboda\source\repos\Kromtec\LegendsViewer\LegendsViewer\Controls\Map\MapPanel.cs:line 1473
   at LegendsViewer.Controls.HTML.BattlePrinter.Print() in C:\Users\Thomas Svoboda\source\repos\Kromtec\LegendsViewer\LegendsViewer\Controls\HTML\BattlePrinter.cs:line 61
   at LegendsViewer.Controls.HTML.HtmlPrinter.GetHtmlPage() in C:\Users\Thomas Svoboda\source\repos\Kromtec\LegendsViewer\LegendsViewer\Controls\HTML\HTMLPrinter.cs:line 161
   at LegendsViewer.Controls.HTML.HtmlControl.GetControl() in C:\Users\Thomas Svoboda\source\repos\Kromtec\LegendsViewer\LegendsViewer\Controls\HTML\HTMLControl.cs:line 50
   at LegendsViewer.Controls.DwarfTabPage.LoadPageControl() in C:\Users\Thomas Svoboda\source\repos\Kromtec\LegendsViewer\LegendsViewer\Controls\DwarfTabPage.cs:line 38
   at LegendsViewer.Controls.DwarfTabPage.NewPageControl(PageControl pageControl) in C:\Users\Thomas Svoboda\source\repos\Kromtec\LegendsViewer\LegendsViewer\Controls\DwarfTabPage.cs:line 33
   at LegendsViewer.Controls.DwarfTabControl.Navigate(ControlOption controlOption, Object navigateObject) in C:\Users\Thomas Svoboda\source\repos\Kromtec\LegendsViewer\LegendsViewer\Controls\DwarfTabControl.cs:line 61
   at LegendsViewer.Controls.Tabs.BaseSearchTab.ListSearch_SelectedIndexChanged(Object sender, EventArgs e) in C:\Users\Thomas Svoboda\source\repos\Kromtec\LegendsViewer\LegendsViewer\Controls\Tabs\BaseSearchTab.cs:line 94
   at System.Windows.Forms.ListView.OnSelectedIndexChanged(EventArgs e)
   at BrightIdeasSoftware.ObjectListView.OnSelectedIndexChanged(EventArgs e)
   at System.Windows.Forms.ListView.WmReflectNotify(Message& m)
   at System.Windows.Forms.ListView.WndProc(Message& m)
   at System.Windows.Forms.NativeWindow.Callback(IntPtr hWnd, Int32 msg, IntPtr wparam, IntPtr lparam)


************** Loaded Assemblies **************
mscorlib
    Assembly Version: 4.0.0.0
    Win32 Version: 4.7.3468.0 built by: NET472REL1LAST_C
    CodeBase: file:///C:/Windows/Microsoft.NET/Framework64/v4.0.30319/mscorlib.dll
----------------------------------------
LegendsViewer
    Assembly Version: 1.20.2.0
    Win32 Version: 1.20.02
    CodeBase: file:///C:/Users/Ralpha/Downloads/LegendsViewer-1.20.02/LegendsViewer-1.20.02/LegendsViewer.exe
----------------------------------------
System.Windows.Forms
    Assembly Version: 4.0.0.0
    Win32 Version: 4.7.3580.0 built by: NET472REL1LAST_C
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Windows.Forms/v4.0_4.0.0.0__b77a5c561934e089/System.Windows.Forms.dll
----------------------------------------
System
    Assembly Version: 4.0.0.0
    Win32 Version: 4.7.3451.0 built by: NET472REL1LAST_C
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System/v4.0_4.0.0.0__b77a5c561934e089/System.dll
----------------------------------------
System.Drawing
    Assembly Version: 4.0.0.0
    Win32 Version: 4.7.3190.0 built by: NET472REL1LAST_C
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Drawing/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Drawing.dll
----------------------------------------
System.Core
    Assembly Version: 4.0.0.0
    Win32 Version: 4.7.3580.0 built by: NET472REL1LAST_C
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Core/v4.0_4.0.0.0__b77a5c561934e089/System.Core.dll
----------------------------------------
System.Configuration
    Assembly Version: 4.0.0.0
    Win32 Version: 4.7.3324.0 built by: NET472REL1LAST_C
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Configuration/v4.0_4.0.0.0__b03f5f7f11d50a3a/System.Configuration.dll
----------------------------------------
System.Xml
    Assembly Version: 4.0.0.0
    Win32 Version: 4.7.3190.0 built by: NET472REL1LAST_C
    CodeBase: file:///C:/Windows/Microsoft.Net/assembly/GAC_MSIL/System.Xml/v4.0_4.0.0.0__b77a5c561934e089/System.Xml.dll
----------------------------------------
ObjectListView
    Assembly Version: 2.9.1.1072
    Win32 Version: 2.9.1.0
    CodeBase: file:///C:/Users/Ralpha/Downloads/LegendsViewer-1.20.02/LegendsViewer-1.20.02/ObjectListView.DLL
----------------------------------------
SevenZipSharp
    Assembly Version: 0.64.3890.29348
    Win32 Version: 0.64.3890.29348
    CodeBase: file:///C:/Users/Ralpha/Downloads/LegendsViewer-1.20.02/LegendsViewer-1.20.02/SevenZipSharp.DLL
----------------------------------------
CommonMark
    Assembly Version: 0.1.0.0
    Win32 Version: 0.1.0.0
    CodeBase: file:///C:/Users/Ralpha/Downloads/LegendsViewer-1.20.02/LegendsViewer-1.20.02/CommonMark.DLL
----------------------------------------
Jdenticon
    Assembly Version: 2.2.1.0
    Win32 Version: 2.2.1.0
    CodeBase: file:///C:/Users/Ralpha/Downloads/LegendsViewer-1.20.02/LegendsViewer-1.20.02/Jdenticon.DLL
----------------------------------------
Jdenticon.Gdi
    Assembly Version: 2.2.1.0
    Win32 Version: 2.2.1.0
    CodeBase: file:///C:/Users/Ralpha/Downloads/LegendsViewer-1.20.02/LegendsViewer-1.20.02/Jdenticon.Gdi.DLL
----------------------------------------

************** JIT Debugging **************
To enable just-in-time (JIT) debugging, the .config file for this
application or computer (machine.config) must have the
jitDebugging value set in the system.windows.forms section.
The application must also be compiled with debugging
enabled.

For example:

<configuration>
    <system.windows.forms jitDebugging="true" />
</configuration>

When JIT debugging is enabled, any unhandled exception
will be sent to the JIT debugger registered on the computer
rather than be handled by this dialog box.
```

**Comment by ralpha** [comment]
Same error in 1.20.03
And also happens for some Conquerings.

**Comment by Kromtec** [comment]
Sorry for answering late. I can not reproduce this with my internal build.
Please upload a zip with your exported files if possible.

**Comment by ralpha** [comment]
I use currently use this map for a DnD campaign I'm running so don't want the world to be public. (as they might find it)
I have send you the files via email. If you did not receive this let me know.

I could not reproduce this bug with other save files I had from the same df version.

Steps to reproduce:
- Look in Warfare -> Battles -> Search
- Select a battle from the list here are some battles that I found give the error:
  - The Violent Clashes
  - The Siege of Conflagration
  - The Attack of Daggers
  - ...
- Error will pop up while loading the info page. (but page will not be visible)

While typing this message and trying to reproduce it on different saves I realized that the files might be changed a little as I used them for tools as well. I for example changed all the files to UTF-8 encoding using:
`$ iconv -f CP437 -t UTF-8 $REGION-legends.xml.old -o $REGION-legends.xml`
Although I was not able to reproduce this error by processing another save similarly.

Using Legends Browser I saw that most of the battles that crash are started by "The Remarkable Dive"
Opening the battle through a link on a page, I do not get an error although the page does not load.

Hope this helps.

**Comment by Kromtec** [comment]
Thx! It was because of battles with outcasts that lived for example in a forest instead of a site. 
I could fix the bug, thanks to your files and the fix will be in the next version.
![image](https://user-images.githubusercontent.com/6030912/78111866-9ca1f000-73fd-11ea-9e86-5c0de769e74d.png)

**Comment by Kromtec** [comment]
fixed in 1.20.04
